src/events/tls.py

```diff
@@ -131,12 +131,16 @@ def _on_certificate_available(self, event: CertificateAvailableEvent):
         if not cert or not private_key:
             logger.error("Missing certificate or private key")
             raise Exception("Missing certificate or private key")
-
-        self.charm.set_status(Status.TLS_NOT_READY)
         # write certificates to disk
         self.charm.tls_manager.write_certificate(cert, private_key)
 
+        # Updating certificates no need to do a rolling restart
+        if self.charm.state.unit_server.tls_state == TLSState.TLS:
```

Comment:
what of the case where the peer-cert was issued but this hook was invoked for the client-cert? what would the state of tls be?